### ISSUE TITLE
Reranker score normalization assumes raw logits, but Cohere-compatible providers may already return probabilities

### ISSUE THREAD

**Issue by badhakker** [vectorize-io/hindsight#1727]
## Summary

Hindsight currently applies sigmoid normalization unconditionally to all reranker provider scores in `hindsight_api/engine/search/reranking.py`.

This works for local SentenceTransformers CrossEncoder models that return raw logits, but it breaks score calibration for providers/endpoints that already return normalized relevance scores/probabilities in `[0, 1]`.

This affects Cohere-compatible `/rerank` endpoints, including llama.cpp Qwen3 rerankers.

## Environment

- Hindsight image: `ghcr.io/vectorize-io/hindsight:latest`
- Hindsight version from image label: `0.6.2`
- Reranker provider: `cohere`
- Custom base URL: llama.cpp `/v1/rerank`
- Model: Qwen3 reranker GGUF served by llama.cpp

Relevant config:

```bash
HINDSIGHT_API_RERANKER_PROVIDER=cohere
HINDSIGHT_API_RERANKER_COHERE_BASE_URL=http://host:9022/v1/rerank
HINDSIGHT_API_RERANKER_COHERE_MODEL=Qwen3-Reranker-4B-Q4_K_M.gguf
HINDSIGHT_API_RERANKER_COHERE_API_KEY=local

### Steps to Reproduce

What happens
llama.cpp Qwen3 reranker returns relevance_score values that are already probabilities.

Example raw llama.cpp response:

{
  "results": [
    {"index": 0, "relevance_score": 0.934695303440094},
    {"index": 1, "relevance_score": 0.005602922290563583},
    {"index": 2, "relevance_score": 0.0016631170874461532}
  ]
}
Hindsight then applies sigmoid again:

sigmoid(0.9347)  -> 0.7180
sigmoid(0.0056)  -> 0.5014
sigmoid(0.00166) -> 0.5004
As a result, irrelevant candidates get scores around 0.5, which makes score thresholds and score interpretation unreliable.


Source references
In Hindsight:

# hindsight_api/engine/search/reranking.py
normalized_scores = [sigmoid(score) for score in scores]
The docs also state that cross-encoders return raw logits:

CE_normalized = 1 / (1 + e^(-raw_logit))
But the Cohere-compatible provider contract uses:

{"results": [{"index": 0, "relevance_score": 0.9}]}
For many APIs, relevance_score is already a normalized score/probability.

In llama.cpp Qwen3 reranker, Qwen3 scores are explicitly softmaxed before being returned:

// softmax for qwen3 reranker
if (arch == LLM_ARCH_QWEN3 || arch == LLM_ARCH_QWEN3VL) {
    cur = ggml_soft_max(ctx0, cur);
}


### Expected Behavior

Expected behavior
Hindsight should support provider-aware or configurable score normalization.

For example:

HINDSIGHT_API_RERANKER_SCORE_NORMALIZATION=sigmoid  # current default
HINDSIGHT_API_RERANKER_SCORE_NORMALIZATION=none     # provider already returns [0,1]
Then:

sigmoid: use current behavior for raw-logit cross-encoders
none: use provider score directly
optionally clamp none mode to [0, 1] for safety
Why this matters
The issue does not necessarily break ranking order because sigmoid is monotonic, but it breaks score calibration:

irrelevant probability 0.005 becomes 0.501
all non-negative probability scores become >= 0.5
score thresholds and trace/debug interpretation become misleading

### Actual Behavior

Proposed fix
Add a config option, defaulting to current behavior for backward compatibility:

HINDSIGHT_API_RERANKER_SCORE_NORMALIZATION=sigmoid
Possible values:

sigmoid
none
identity
passthrough
Implementation location:

hindsight_api/engine/search/reranking.py
Pseudo-code:

mode = os.getenv("HINDSIGHT_API_RERANKER_SCORE_NORMALIZATION", "sigmoid").lower()

if mode == "sigmoid":
    normalized_scores = [sigmoid(score) for score in scores]
elif mode in ("none", "identity", "passthrough"):
    normalized_scores = [float(score) for score in scores]
else:
    raise ValueError(...)
This keeps backward compatibility while allowing Cohere-compatible probability-based rerankers to work correctly.

### Version

_No response_

### LLM Provider

None

**Comment by nicoloboschi** [comment]
fixed already by https://github.com/vectorize-io/hindsight/pull/1512 
will be shipped in 0.6.3 tomorrow
